### PR TITLE
Feature/trim time shifted data

### PR DESCRIPTION
**Pull request by EddyCMWF**
Option to trim data if time_shift is applied

### CODE REVIEW

File: earthkit/transforms/aggregate/temporal.py [hunk 1]
```diff
@@ -379,10 +379,13 @@ def daily_reduce(
...
         default behaviour is to deduce time dimension from
         attributes of coordinates, then fall back to `"time"`.
-    time_shift : (optional) timedelta or dict
+    time_shift : (optional) None, timedelta or dict
         A time shift to apply to the data prior to calculation, e.g. to change the local time zone.
         It can be provided as any object that can be understood by `pandas.Timedelta`, a dictonary is passed
-        as kwargs to `pandas.Timedelta`
+        as kwargs to `pandas.Timedelta`. Default is None.
+    remove_partial_periods : bool
+        If True and time_shift is not Null, the first and last time steps are removed to ensure equality in
```
Comment (JamesVarndell) [inline earthkit/transforms/aggregate/temporal.py]: I would suggest a clarification of language
```suggestion
        If True and a time_shift has been applied, the first and last time steps are removed to ensure equality in
```

File: earthkit/transforms/tools.py [hunk 2]
```diff
@@ -65,7 +66,17 @@ def wrapper(
 
             dataarray = dataarray.assign_coords({time_dim: time_coord})
 
-        return func(dataarray, *args, time_dim=time_dim, **kwargs)
+        result = func(dataarray, *args, time_dim=time_dim, **kwargs)
+
+        # If we want only full days then chop off the first and last day if we have time_shift
+        if (
+            remove_partial_periods
+            and time_shift is not None
```
Comment (JamesVarndell) [inline earthkit/transforms/tools.py]: Should this also check for equivalently null `time_shift` values like `0`?

File: earthkit/transforms/tools.py [hunk 3]
```diff
@@ -65,7 +66,17 @@ def wrapper(
...
+            and isinstance(result, (xr.Dataset, xr.DataArray))
+        ):
+            return result.isel({time_dim: slice(1, -1)})
```
Comment (JamesVarndell) [inline earthkit/transforms/tools.py:73]: Is this safe? What if the time shift is 24 hours? Should we double-check that each end chunk is definitely not a complete time chunk?

File: earthkit/transforms/tools.py [hunk 4]
```diff
@@ -43,6 +43,7 @@ def wrapper(
         *args,
         time_dim: str | None = None,
         time_shift: dict | str | pd.Timedelta | None = None,
+        trim_shifted: bool = False,
```
[the remark this answers is not in the document]
Comment (EddyCMWF) [reply earthkit/transforms/tools.py]: resolved as suggested

File: earthkit/transforms/aggregate/temporal.py (hunk 1 above)
Comment (EddyCMWF) [reply earthkit/transforms/aggregate/temporal.py]: done

File: earthkit/transforms/tools.py (hunk 3 above)
Comment (EddyCMWF) [reply earthkit/transforms/tools.py:73]: Well this is why I left it as a switch, I'm not sure we could make a safe detector, the time is not necessarily contiguous, and I don't want to make that assumption

File: earthkit/transforms/tools.py (hunk 2 above)
Comment (EddyCMWF) [reply earthkit/transforms/tools.py]: I now check if time_shift is, so None, 0, and timedelta(0) all prevent trimming

**Pull request merged**